core/src/main/java/io/cucumber/core/plugin/JSONFormatter.java

```diff
@@ -381,16 +382,16 @@ private void addEmbeddingToHookMap(byte[] data, String mimeType) {
         if (result.getErrorMessage() != null) {
             resultMap.put("error_message", result.getErrorMessage());
         }
-        if (result.getDuration() != 0) {
-            resultMap.put("duration", result.getDuration());
+        if (!result.getTimeDuration().isZero()) {
+            resultMap.put("duration", result.getTimeDuration().toNanos());
         }
         return resultMap;
     }
     
-    private String getDateTimeFromTimeStamp(long timeStampMillis) {
+    private String getDateTimeFromTimeStamp(Instant instant) {
         SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXXX");
         sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
         
-        return sdf.format(new Date(timeStampMillis));
+        return sdf.format(new Date(instant.toEpochMilli()));
```

Comment:
It would be cleaner to directly format the instant. Java 8 has a new formatter for temporals.